File: src/ft.transaction-management/ft.transaction-management.Domain/Entities/Transaction.cs
using System;
using ft.transaction_management.Domain.Common;

namespace ft.transaction_management.Domain.Entities;

public class Transaction : BaseDomainEntity
{
    public string? Status { get; set; }
    public int Quantity { get; set; }
    public string? Type { get; set; }
    public DateTime Date { get; set; }
    public string? AccountNumber { get; set; }
    public decimal Amount { get; set; }
    public string? Region { get; set; }
    public int? CategoryId { get; set; }
    public int? NumberOfItems { get; set; }
    public Category? Category { get; set; }
    public double? CostPerItem { get; set; }
    public string? Description { get; set; }
    public string? PaymentMethod { get; set; }
}

File: src/ft.web-ui/src/types/transaction.ts
import { Category } from './categories'

export type Transaction = {
  id: number
  date: Date
  type: string
  amount: number
  status: string
  region?: string | null
  description?: string | null
  paymentMethod?: string | null
  quantity?: number
  unitPrice?: number
  accountNumber?: string
  categoryId: number | null
  category: Category | null
}
